astropy/convolution/src/boundary_padded.c

```diff
@@ -1,539 +0,0 @@
-                g, kernel_shape[0], kernel_shape[1],
-                nan_interpolate, n_threads);
-    else if (n_dim == 3)
-        convolve3d_padded_boundary_c(result, f,
-                        image_shape[0], image_shape[1], image_shape[2],
-                        g, kernel_shape[0], kernel_shape[1], kernel_shape[2],
-                        nan_interpolate, n_threads);
-    else
-        assert(0); // Unimplemented: n_dim > 3
-}
-
-/*-------------------------PERFORMANCE NOTES--------------------------------
- * The function wrappers below are designed to take advantage of the following:
- * The preprocessor will inline convolve<N>d_padded_boundary(), effectively
- * expanding the two logical branches, replacing nan_interpolate
- * for their literal equivalents. The corresponding conditionals
- * within these functions will then be optimized away, this
- * being the goal - removing the unnecessary conditionals from
- * the loops without duplicating code.
- *--------------------------------------------------------------------------
- */
-
-void convolve1d_padded_boundary_c(DTYPE * const result,
-        const DTYPE * const f, const size_t nx,
-        const DTYPE * const g, const size_t nkx,
-        const bool nan_interpolate,
-        const unsigned n_threads)
-{
-#ifdef NDEBUG
-    if (!result || !f || !g)
-        return;
-#else
-    assert(result);
-    assert(f);
-    assert(g);
-#endif
-
-    if (nan_interpolate)
-        convolve1d_padded_boundary(result, f, nx, g, nkx, true, n_threads);
-    else
-        convolve1d_padded_boundary(result, f, nx, g, nkx, false, n_threads);
-}
-
-void convolve2d_padded_boundary_c(DTYPE * const result,
-        const DTYPE * const f, const size_t nx, const size_t ny,
-        const DTYPE * const g, const size_t nkx, const size_t nky,
-        const bool nan_interpolate,
-        const unsigned n_threads)
-
-{
-#ifdef NDEBUG
-    if (!result || !f || !g)
-        return;
-#else
-    assert(result);
-    assert(f);
-    assert(g);
-#endif
-
-    if (nan_interpolate)
-        convolve2d_padded_boundary(result, f, nx, ny, g, nkx, nky, true, n_threads);
-    else
-        convolve2d_padded_boundary(result, f, nx, ny, g, nkx, nky, false, n_threads);
-}
-
-void convolve3d_padded_boundary_c(DTYPE * const result,
-        const DTYPE * const f, const size_t nx, const size_t ny, const size_t nz,
-        const DTYPE * const g, const size_t nkx, const size_t nky, const size_t nkz,
-        const bool nan_interpolate,
-        const unsigned n_threads)
-{
-#ifdef NDEBUG
-    if (!result || !f || !g)
-        return;
-#else
-    assert(result);
-    assert(f);
-    assert(g);
-#endif
-
-    if (nan_interpolate)
-        convolve3d_padded_boundary(result, f, nx, ny, nz, g, nkx, nky, nkz, true, n_threads);
-    else
-        convolve3d_padded_boundary(result, f, nx, ny, nz, g, nkx, nky, nkz, false, n_threads);
-}
-
-/*-------------------------PERFORMANCE NOTES--------------------------------
```

Comment:
My understanding was this is no longer relevant - the kernel can't overstep the image since the array is padded using the kernel size? Thus the performance notes in the non-padded case are enough? (in fact, this was already the case with your original PR, I just missed it at the time)

Reply:
It's just this is the rationale for padding in the first place. I think that it should persist somewhere other than in old commit space. Either in the code or I can copy it out to a comment in #7293.

Reply:
Ok sounds good - I'll copy it over